build-system/tasks/runtime-test/helpers.js

```diff
@@ -156,15 +156,12 @@ function maybePrintArgvMessages() {
 async function karmaBrowserComplete_(browser) {
   const result = browser.lastResult;
   result.total = result.success + result.failed + result.skipped;
-  // Initially we were reporting an error with reportTestErrored() when zero tests were detected (see #16851),
-  // but since Karma sometimes returns a transient, recoverable state, we will
-  // print a warning without reporting an error to the github test status. (see #24957)
+  // This used to be a warning with karma-browserify. See #16851 and #24957.
+  // Now, with karma-esbuild, this is a fatal error. See #34040.
   if (result.total == 0) {
```

Comment:
Turns out that we can't just throw here because the Karma server will stall when it encounters an unhandled exception. The only clean way is to print an error message and exit. This is fine given that this is an unrecoverable fatal error.

![image](https://user-images.githubusercontent.com/26553114/116319838-349b1600-a785-11eb-8036-4d718866c56d.png)

Reply:
This was addressed by b607ff8.